apps/web/src/components/auth/use-accept-invite.hook.ts

```diff
@@ -14,20 +19,29 @@ export function useAcceptInvite() {
     string
   >((tokenItem) => api.post(`/v1/invites/${tokenItem}/accept`, {}));
 
-  const submitToken = async (token: string, refetch = false): Promise<boolean> => {
-    const responseInvite = await mutateAsync(token);
-
-    if (!responseInvite) return false;
-
-    setToken(responseInvite);
-    if (refetch) {
-      await queryClient.refetchQueries({
-        predicate: (query) => query.queryKey.includes('/v1/organizations'),
+  const submitToken = useCallback(async (token: string, invitationToken: string, refetch = false) => {
+    try {
+      // just set the header, user is logged in after token is submitted
+      applyToken(token);
+      const newToken = await mutateAsync(invitationToken);
+      if (!newToken) return;
+
+      setToken(newToken, refetch);
```

Comment:
The `/v1/invites/${tokenItem}/accept`endpoint requires the JWT, so we call `applyToken` just to set the header, then the result is a new JWT token sent back, which we update in the app with `setToken`.

I haven't used the `setToken` instead of `applyToken` because the first one triggers additional calls to our API, which we don't want to do.